### PR TITLE
Improved device() method

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Make idist.device() output uniform : torch.device

Check list:
* [x] New tests are added (if a new feature is added)
* [x] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by vfdev-5** [comment]
We can ignore codecov for a moment as we do not merge all reports.

File: ignite/distributed/comp_models/base.py [hunk 1]
```diff
@@ -143,9 +143,6 @@ def get_num_nodes(self) -> int:
     def get_node_rank(self) -> int:
         return 0
 
-    def is_distributed(self) -> bool:
```
Comment (erip) [inline ignite/distributed/comp_models/base.py:146]: Question: why was this removed? There may be a good reason, but I think now the contract enforced by the interface is more implicit.

Comment (vfdev-5) [reply ignite/distributed/comp_models/base.py:146]: @erip it was discussed in a gdoc about idist API and we thought that it is sort of redundant and what to call `distributed`... So, now we check `get_world_size() > 1`.
